docs/recipes/upgrading/v3-to-v4.md

```diff
@@ -0,0 +1,101 @@
+
+### Terminology for this Recipe:
+
+* ***"Overwrite"***: Take the file from the "new" V4 template and copy it over the equivalent file in your existing project.
+* ***"Delete"***: Delete the file from your existing project. It is no longer required.
+* ***"Add"***: Add the file to your existing project. It is a new file added in V4.
+
+#### 1. Install the v4 template
+Download and install the latest SAFE Stack V3 template by running the following command:
+
+```bash
+dotnet new -i SAFE.Template
+```
+
+#### 2. Create a v4 project
+Create a new SAFE project in the `safetemp` folder. We will use this as a basis for our conversion.
+
+```fsharp
+dotnet new SAFE -o safetemp
+```
+
+#### 3. Branch your code
+We advise committing or stashing any unsaved changes to your code and making a new branch to perform the upgrade.
+
+You can then test it in isolation and then safely merge back in.
+
+#### 4. Update dotnet tools
+This file lists any custom dotnet tools used.
+
+* **Overwrite** the `.config/dotnet-tools.json` file.
+
+> **Important**! If you have installed extra dotnet tools, you will need to add them back in manually.
+
+#### 5. Update global.json 
+
+* **Overwrite** the `global.json` file
+
+#### 6. Update Paket dependencies
+* **Overwrite** the `paket.dependencies` file in the root of the solution.
+* **Overwrite** the `paket.lock` file.
+* **Overwrite** all `paket.references` files.
+
+**Important** If you have installed extra NuGet packages, you will need to add them back in manually to the dependencies and references files.
+
+Run paket to update project references:
+
+```bash
+dotnet paket install
+```
+
+* If you *have* added any extra NuGet packages, this command will also generate a new `paket.lock` file.
+
+#### 7. Update the npm dependancies 
+* **Overwite** the `package.json` file
+* **Overwite** the `package-lock.json` file
+
+**Important** If you have installed extra npm packages, you will need to add them back in manually to the dependencies.
+
+#### 8. Update .gitignore 
+* **Overwite** the `.gitignore` file in the root of the solution
+
+#### 9. Update the build process
+* **Overwite** the `Build.fs` file
+* **Overwite** the `Build.fsproj` file
+
+**Important** If you have made any modifications to the build script e.g. extra targets, you will need to add them back in manually.
+
+#### 10. Update the webpack config
+* **Overwrite** the `webpack.config.js` file.
+* **Delete** the `webpack.tests.config.js` file
+
+**Important** If you have made any modifications to the webpack file, you will need to apply them back in manually.
+
+#### 11. Update TargetFramework in all projects
+* **Overwite** the `Client.Tests.fsproj` file
+* **Overwite** the `Server.Tests.fsproj` file
+* **Overwite** the `Shared.Tests.fsproj` file
+* **Overwite** the `Client.fsproj` file
+* **Overwite** the `Server.fsproj` file
+* **Overwite** the `Shared.fsproj` file
```

Comment:
Do we need to say about copying across any modifications to the file? Would it be better to get them just to change the TargetFramework property?

Reply:
Yeah perhaps - I was just really following the precedent from the 2->3 migration of either remove or replace file instructions, with caveats that if you have edited the originals, you need to take that into consideration.

I'd need to check that doing that is all that is necessary but sounds feasible

Reply:
Just checked the diff and it is the only change in the files so yeah, I'll do that